app/services/user_destroyer.rb

```diff
@@ -35,6 +35,10 @@ def destroy(user, opts = {})
       category_topic_ids = Category.where("topic_id IS NOT NULL").pluck(:topic_id)
 
       if opts[:delete_posts]
+        DiscoursePluginRegistry.user_destroyer_on_content_deletion_callbacks.each do |cb|
+          cb.call(user, @guardian, opts)
```

Comment:
I suppose here: `@guardian.user === user` right? If yes, I would just pass the user object around. You can always do `Guardian.new(user)` if you need guardian later.

Reply:
ok :+1: